presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -470,6 +475,104 @@ public DeleteNode plan(Delete node)
         return new DeleteNode(idAllocator.getNextId(), builder.getRoot(), new DeleteTarget(handle, metadata.getTableMetadata(session, handle).getTable()), rowId, outputs);
     }
 
+    public UpdateNode plan(Update node)
+    {
+        Table table = node.getTable();
+        TableHandle handle = analysis.getTableHandle(table);
+
+        Map<String, ColumnHandle> dataColumnHandles = metadata.getColumnHandles(session, handle);
+        List<String> targetColumnNames = node.getAssignments().stream()
+                .map(assignment -> assignment.getName().getValue())
+                .collect(toImmutableList());
+
+        // Create an UpdateAssignments list in table column order
+        ImmutableList.Builder<Expression> orderedExpressionBuilder = ImmutableList.builder();
+        for (String columnName : dataColumnHandles.keySet()) {
+            int index = targetColumnNames.indexOf(columnName);
+            if (index < 0) {
+                continue;
+            }
+            orderedExpressionBuilder.add(node.getAssignments().get(index).getValue());
+        }
+        List<Expression> orderedExpressions = orderedExpressionBuilder.build();
+
+        // create table scan
+        RelationPlan relationPlan = new RelationPlanner(analysis, symbolAllocator, idAllocator, lambdaDeclarationToSymbolMap, metadata, outerContext, session, recursiveSubqueries)
+                .process(table, null);
+
+        PlanBuilder builder = newPlanBuilder(relationPlan, analysis, lambdaDeclarationToSymbolMap);
+
+        if (node.getWhere().isPresent()) {
+            builder = filter(builder, node.getWhere().get(), node);
+        }
+
+        builder = builder.appendProjections(orderedExpressions, symbolAllocator, idAllocator);
+
+        PlanAndMappings planAndMappings = coerce(builder, orderedExpressions, analysis, idAllocator, symbolAllocator, typeCoercion);
+        builder = planAndMappings.getSubPlan();
+        ProjectNode projectNode = (ProjectNode) builder.getRoot();
+
+        List<Symbol> expressionSymbols = orderedExpressions.stream()
+                .map(expression -> requireNonNull(planAndMappings.get(expression), "planAndMappings.get(expression) is null"))
```

Comment:
remove `requireNonNull` (if anything, it should be a `verify()`, but I think it's overkill) and use Collectors.toImmutableList() below.

Reply:
`requireNonNull` removed, but the list needs to be mutable in order to add the rowId symbol at the end.